model-optimizer/extensions/back/add_outputs_recursive.py

```diff
@@ -45,7 +65,10 @@ def ti_infer(step_node, port_num):
     port_num = port_num - len(step_node.in_ports())
 
     # find out iterations count for TensorIterator to set output shape correctly
-    iterations_count = TensorIterator.find_iterations_count_for_output(step_node, found_rec)
+
+    iterations_count = get_iterations_count_from_output_record(found_rec)
+    if iterations_count is None:
+        iterations_count = TensorIterator.find_iterations_count_for_output(step_node)
```

Comment:
please compute iteration number in the single function but the concrete output record is not needed. We need to walkthrough all input and output records to compute iteration number

Reply:
We have discussed it. Output can include only some iteration results theoretically. So we will not analyze all outputs to find out iterations count. 

Analysis of output record where we add iterations count is required because it can include not all iterations or not include it at all.